src/index.ts

```diff
@@ -478,16 +478,17 @@ function createTranslator(
     const existingKeys = destinationFile
       ? Object.keys(destinationFile.content)
       : [];
-    const templateStrings = Object.keys(sourceFile.content);
+    const templateStrings = Object.entries(sourceFile.content);
     const stringsToTranslate = templateStrings
-      .filter((key) => overwrite || !existingKeys.includes(key) || cacheDiff.includes(key))
-      .map((key) => ({
-        key,
-        value: sourceFile.type === 'key-based' ? sourceFile.content[key] : key,
+      .filter((s) => overwrite || !existingKeys.includes(s[0]) || cacheDiff.includes(s[0]) ||
+        (typeof s[1] == 'string' && destinationFile?.content[s[0]]?.trim() == ""))
```

Comment:
How about:
```suggestion
        (typeof sourceFile.content[key] == 'string' && destinationFile?.content[s[0]]?.trim() == ""))
```

Then you don't need to change the rest. Does that work?

Reply:
Yes, thanks a lot for the suggestion!